android/src/main/java/com/epicshaggy/biometric/NativeBiometric.java

```diff
@@ -283,9 +339,55 @@ private String decryptString(String stringToDecrypt, String KEY_ALIAS) throws Ge
     }
 
 
+    private KeyPair generateKeyPair() throws GeneralSecurityException {
+        //Create the start and expiry date for the key
+        // TODO: confirm the expiration date
+        GregorianCalendar startDate = new GregorianCalendar();
+        GregorianCalendar endDate = new GregorianCalendar();
+        endDate.add(Calendar.YEAR, 1);
+
+        //Create RSA key pair and store it in Android Keystore
+        KeyPairGenerator generator = KeyPairGenerator.getInstance(KeyProperties.KEY_ALGORITHM_RSA, ANDROID_KEY_STORE);
+
+        //Creating the key pair with sign and verify purposes
+        generator.initialize(new KeyGenParameterSpec.Builder(DEFAULT_KEY,
+                KeyProperties.PURPOSE_SIGN | KeyProperties.PURPOSE_VERIFY).
+                setCertificateSerialNumber(BigInteger.valueOf(777)).                        //Serial number used for the self-signed certificate of the generated key pair, default is 1
+                        setCertificateSubject(new X500Principal("CN=DefaultKey")).    //Subject used for the self-signed certificate of the generated key pair, default is CN=fake
+                        setDigests(KeyProperties.DIGEST_SHA256).                            //Set of digests algorithms with which the key can be used
+                        setSignaturePaddings(KeyProperties.SIGNATURE_PADDING_RSA_PKCS1).    //Set of padding schemes with which the key can be used when signing/verifying
+                        setCertificateNotBefore(startDate.getTime()).                       //Start of the validity period for the self-signed certificate of the generated, default Jan 1 1970
+                        setCertificateNotAfter(endDate.getTime()).                          //End of the validity period for the self-signed certificate of the generated key, default Jan 1 2048
+                        setUserAuthenticationRequired(true).                                //Sets whether this key is authorized to be used only if the user has been authenticated, default false
+                        setUserAuthenticationValidityDurationSeconds(30).                   //Duration(seconds) for which this key is authorized to be used after the user is successfully authenticated
+                        build());
+
+        return generator.genKeyPair();
+    }
+
+    private KeyPair getKeyPair() throws GeneralSecurityException, IOException {
+        KeyPair keyPair = null;
+        KeyStore keystore = getKeyStore();
+
+        Key key = keystore.getKey(DEFAULT_KEY, null);
+
+        if (key instanceof PrivateKey && key != null) {
```

Comment:
```suggestion
        if (key != null && key instanceof PrivateKey) {
```
Check null first before instanceof